hail/python/hail/ir/table_ir.py

```diff
@@ -934,8 +934,9 @@ def __init__(self, child, config):
 
     def _handle_randomness(self, uid_field_name):
         if uid_field_name is not None:
-            raise FatalError('TableToTableApply does not support randomness in consumers')
-        child = self.child.handle_randomness(None)
+            if self.config['name'] != 'TableFilterPartitions':
+                raise FatalError(f'TableToTableApply({self.config["name"]}) does not support randomness in consumers')
+        child = self.child.handle_randomness(uid_field_name)
```

Comment:
why did we switch to using uid_field_name from None?

Reply:
TableFilterPartitions is the only TableToTableApply where we do support passing through random UIDs. Others trigger an error here.